packages/block-editor/src/components/block-mobile-toolbar/block-actions-menu.native.js

```diff
@@ -325,23 +325,19 @@ export default compose(
 			canInsertBlockType,
 			getTemplateLock,
 		} = select( blockEditorStore );
-		const normalizedClientIds = Array.isArray( clientIds )
-			? clientIds
-			: [ clientIds ];
-		const block = getBlock( normalizedClientIds );
-		const blockName = getBlockName( normalizedClientIds );
+		const block = getBlock( clientId );
+		const blockName = getBlockName( clientId );
 		const blockType = getBlockType( blockName );
 		const blockTitle = blockType?.title;
-		const firstClientId = normalizedClientIds[ 0 ];
+		const firstClientId = clientId;
 		const rootClientId = getBlockRootClientId( firstClientId );
 		const blockOrder = getBlockOrder( rootClientId );
 
 		const firstIndex = getBlockIndex( firstClientId );
-		const lastIndex = getBlockIndex(
-			normalizedClientIds[ normalizedClientIds.length - 1 ]
-		);
+		const isFirst = firstIndex === 0;
+		const isLast = clientId === blockOrder.length - 1;
```

Comment:
This feels off - comparing the client ID (which I expect to be a unique string) to the index of the blocks. Did we mean to use `getBlockIndex()` here as well?

As a matter of fact, since we no longer work with an array of client IDs, shouldn't we just be using `getBlockIndex( clientId )` for the last index, just like we are already doing for the first index?

Reply:
It was indeed off 😅  thank you for spotting that, I've updated this logic in https://github.com/WordPress/gutenberg/pull/46699/commits/82c2b8598dcdba58f59eb5da820c6f5544f49559#diff-5977ed5d4a2d269bae979099b537fe8daef530d2513e09ae1fedabb374e82c52